sql-to-dbsp-compiler/lib/sqllib/src/casts.rs

```diff
@@ -404,23 +399,14 @@ pub fn cast_to_decimalN_fN(value: Option<F32>, precision: u32, scale: u32) -> Op
 
 #[inline]
 pub fn cast_to_decimalN_s(value: String, precision: u32, scale: u32) -> Option<Decimal> {
-    let result = match value.trim().parse() {
-        Err(_) => Some(Decimal::zero()),
-        Ok(x) => Some(x),
-    };
+    let result = Some(value.trim().parse().unwrap());
```

Comment:
so this panics (unwrap) instead of returning None?

Reply:
that is what our current spec says, and it's consistent with the rest of SQL

Reply:
so this was in fact a bug (none of the 4 resolved)